docs/RFCS/20220608_split_jobs_system_tables.md

```diff
@@ -0,0 +1,279 @@
+
+a) **the API will not allow transactional interactions with job_control which could block**, and
+
+b) **there may be many rows per job in job_info. **
+
+
+## Detailed Design
+
+### system.job_control
+
+#### Description
+
+The system.job_control table stores the core state of the jobs system; it defines what jobs are currently running, have run, need to start running or need to stop running. 
+
+This table is _exclusively_ used by the jobs system itself; no user-controlled or job-controlled transactions are allowed in this table. This avoids a misbehaving job or user from blocking the jobs system and its ability to continue to manage other jobs.
+
+Job-controlled transactions have historically shown two main forms of misbehaving in this table: taking too long to commit updates, or committing large updates too often, with both causing contention and the latter causing accumulation of mvcc garbage. Both of these are mitigated by providing jobs a separate table described below in which to read and write its own information -- using transactions that block for as long as it wishes and spread across many small rows if needed -- leaving this control table to be only those fields needed by the job control system itself.
+
+Thus this table only contains fields common to all jobs, not job specific protos or arguments. It specifies that a particular job ID, with a particular type, needs to be started, stopped, etc. The jobs system then uses that information to find the relevant implementing code -- i.e. the resumer registered for that type -- and call the relevant method -- e.g. Resume, OnFailOrCancel, etc -- informing it of the ID it should be resuming/cancelling/etc. The job’s implementation of those methods are then responsible for retrieving any job-specific details, such as which files it is importing or what tables it is backing up to where, by looking for the applicable rows it or its planning placed in the job_info table.
+
+While this table specifies the _type_ of each job, it does not store any job-type specific information; i.e. while it indicates that job 123 is a BACKUP job and is ready to run, it does _not_ indicate what job 123 is backing up or how much it has completed; that information must be stored elsewhere, as it is often determined or mutated using explicit transactions. 
+
+#### Schema
+
+```
+    CREATE TABLE system.job_control (
+    	id			INT PRIMARY KEY,
+    	job_type		STRING, 
+    	description 	STRING, 		# human-readable description
+    	created 		TIMESTAMPTZ,
+    	created_by_type	STRING,
+    	created_by_id	INT,
+     	state STRING NOT NULL, 		# control state enum, e.g. running, pause-requested, etc
+    	last_state_change TIMESTAMPTZ,
+
+
+    	# claim/backoff fields determine who should/when to try execute this job
+    	claim_session_id BYTES,
+    claim_instance_id INT,
+    	backoff_num_attempts INT, 
```

Comment:
This column might not be required: I believe ideally we'd let the job itself manage its own retries and any backoff between them it thinks it should add, storing that state in its own info rows, so the backoff at the jobs system level would just be if the node running it crashed before the invoked job-specific function returned, then we should wait some interval before trying again to avoid a job of death crashing all nodes in quick succession.